edb/server/compiler_pool/pool.py

```diff
@@ -452,6 +452,19 @@ def sync_worker_state_cb(
 
         return preargs, callback
 
+    async def _acquire_worker(self):
+        while (
+            worker := await self._workers_queue.acquire()
+        ).get_pid() not in self._workers:
+            # The worker was disconnected; skip to the next one.
+            pass
+        return worker
```

Comment:
Can all workers in the queue be disconnected?  If so, then this potentially still leaks a disconnected worker (the last one).

Reply:
> Can all workers in the queue be disconnected?

Yes!

> this potentially still leaks a disconnected worker (the last one).

But the last one in the queue won't escape the `.get_pid() not in self._workers` check? `_workers_queue.acquire()` will block after that, until a worker whose pid is in `self._workers` is found 🤔

Reply:
Oh, right. `acquire()` blocks.